src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -299,17 +762,103 @@ IMF_STD_ATTRIBUTE_DEF (framesPerSecond, FramesPerSecond, Rational)
 
 
 //
-// multiView -- defines the view names for multi-view image files.
+// multiView -- defines the view names for multi-view image files
+//
 // A multi-view image contains two or more views of the same scene,
 // as seen from different viewpoints, for example a left-eye and
 // a right-eye view for stereo displays.  The multiView attribute
 // lists the names of the views in an image, and a naming convention
 // identifies the channels that belong to each view.
 //
+// Files conformant to SMPTE ST 2065-4 ("ACES Container File Format")
+// which have a multiView attribute will always have the two strings
+// "right" and "left" in that order. When an ACES Container file is
+// used to hold a stereoscopic image, this attribute will be present.
+//
 // For details, please see header file ImfMultiView.h
 //
 
-IMF_STD_ATTRIBUTE_DEF (multiView , MultiView, StringVector)
+IMF_STD_ATTRIBUTE_DEF (multiView, MultiView, StringVector)
+
+
+//
+// framelines -- set of rectangles used for creative purposes, e.g. to
+// indicate safe areas, cropping for derivative image sequences, etc
+//
+// The way in which the format encodes these rectangles is not defined;
+// the encoding might be camera-vendor-specific, or it might be formally
+// or informally standardized (e.g. an ASC Frameline Decision List)
+//
+
+IMF_STD_ATTRIBUTE_DEF (framelines, Framelines, std::string)
+
+
+//
+// imageRotation -- rotation of the image in degrees, in the closed interval
+// [-45, +45], with the rotation around the center of the display window, a
+// positive value specifying a rotation from x-axis to y-axis
+//
+
```

Comment:
For rotations beyond  +/- 45 degrees, is the expectation that the other angles are accomplished via possibly negative scaling of the image itself? Is there any reason to restrict imageRotation at all? It seems to introduce a reliance on related values, and could introduce unexpected results, if either rotation or scale were modified independently. Perhaps all the geometric attributes belong together in the file.

Reply:
ImageRotation seems kind of crazy to me, and as I will show in a minute, it did to Florian as well.

I found a pre-CD version of 2065-4 from March 8, 2012. As you might recall, it had become clear that all cinema acquisition would be done in stereo on every meaningful production, forever forward (this was a couple years before the same thing was said about high frame rate). Thus the early 2012 spec was larded with provisions for 3D. There was an `orientation` attribute that was dropped a week or two later; look on it and wonder:

![image](https://user-images.githubusercontent.com/1134845/103466847-d8644000-4d16-11eb-89ff-acc8a73a5d53.png)

There are some classic pieces of email between Lars and Florian re: imageOrientation and imageRotation around this time. I can't quote them at length without violating SMPTE rules, but to paraphrase him just slightly, "imageOrientation changes the mapping of the data window onto the image; and rotating 90º would map the yMin row onto the first column of data". Florian notes in a response that we're now talking about resampling the image data, with some unspecified interpolation method; that this breaks compatibility with OpenEXR; and that we should drop the attribute. Later he seemed to cooperate and supported a version with rotations up to 90º but since the imageOrientation attribute had been dropped at that point, I don't see why he relented.

Though I will ask Lars _et al._ over on the SMPTE 2065-4 revision working group, my personal thoughts at this point are that (a) we should not put imageRotation into the reference implementation, (b) perhaps it should be deprecated in the published 2065-4:202x version of the standard and (c) I should really go over all the attributes, and any that seem to have even a slight bad smell, I should check my saved email.

Suggested resolution: drop the imageRotation attribute from the reference implementation. We can entertain arguments from someone who actually has used it, if such a person exists, at any later date. Would that resolve your issue?

Reply:
Yes, drop it please :) It seems like a baroque specification for a display window basis transform, with the related values coyly distributed amongst rotation, scale, strides, and who knows what else, ambiguities about sampling and transformation order; and lacks as Florian noted a specification for resampling. Sampling and coordinate transforms are hard enough without a MENSA puzzle in the mix.